### PR TITLE
Don't encode build context path on Windows

### LINKED ISSUE #3870: docker-compose cannot handle unicode file paths on Windows
I've just hit this error:

```
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "compose\cli\main.py", line 61, in main
  File "compose\cli\main.py", line 113, in perform_command
  File "compose\cli\main.py", line 219, in build
  File "compose\project.py", line 300, in build
  File "compose\service.py", line 727, in build
  File "site-packages\docker\api\build.py", line 54, in build
  File "site-packages\docker\utils\utils.py", line 103, in tar
  File "tarfile.py", line 2006, in add
  File "tarfile.py", line 1878, in gettarinfo
WindowsError: [Error 123] The filename, directory name, or volume label syntax is incorrect: 'C:\\[redacted]\\?.jpg'
```

The actual file name is `∆.jpg` (don't ask why). This looks like the same issue in http://stackoverflow.com/questions/11545185/handling-utf-filenames-in-python

### LINKED ISSUE #4055: docker-compose up can't run my app because filename contains special character å
I am on Windows 7 using docker-compose version 1.8.1, build 004ddae.

My file has a name of  **blå.jpg** and docker-compose up fails as below:

$ docker-compose up -d
Building web
Traceback (most recent call last):
  File "<string>", line 3, in <module>
  File "compose\cli\main.py", line 62, in main
  File "compose\cli\main.py", line 114, in perform_command
  File "compose\cli\main.py", line 835, in up
  File "compose\project.py", line 382, in up
  File "compose\service.py", line 305, in ensure_image_exists
  File "compose\service.py", line 727, in build
  File "site-packages\docker\api\build.py", line 55, in build
  File "site-packages\docker\utils\utils.py", line 95, in tar
  File "tarfile.py", line 2007, in add
  File "tarfile.py", line 1879, in gettarinfo
**WindowsError: [Error 2] The system cannot find the file specified: 'C:\Users\U
ser\project\image\bla\xb0.jpg'**
docker-compose returned -1

Any suggestion?

### LINKED ISSUE #4357: Cannot appropriately process non-english characters such as Chinese characters
E.g., The **context** path refers to the project path, which contains Chinese characters. The docker compose resolve these characters as escape characters, which will result in error when composing.
```
version: '2.0'
services:
  service_0:
   build:
      args:
        source: obj/Docker/empty/
        context: "c:\\users\\user1\\documents\\visual studio 2017\\Projects\\webproject\\
        \u8FC7\u4E86\u5C31\u5403\u996D"
        dockerfile: Dockerfile
```

### PR DESCRIPTION
**Pull request by shin-**
Fixes #4357 
Fixes #4055 
Fixes #3870

### CODE REVIEW

File: compose/service.py [hunk 1]
```diff
@@ -769,9 +770,9 @@ def build(self, no_cache=False, pull=False, force_rm=False):
 
         build_opts = self.options.get('build', {})
         path = build_opts.get('context')
-        # python2 os.path() doesn't support unicode, so we need to encode it to
-        # a byte string
-        if not six.PY3:
+        # python2 os.stat() doesn't support unicode on some UNIX, so we
+        # encode it to a bytestring to be safe
+        if not six.PY3 and not IS_WINDOWS_PLATFORM:
```
Comment (BYK) [inline compose/service.py]: Are you sure this fixes the issue? Because I'm using Python 3 and the source from Git directly and still having this issue.

Comment (shin-) [inline compose/service.py]: Please create a new issue with all the information (OS, full stacktrace, Docker & Compose version...)

**Comment by HuanhuanSunMSFT** [comment]
Any ETA on when will a docker compose with this fix be included in Docker for Windows stable version?

**Comment by shin-** [comment]
@HuanhuanSunMSFT It's been there for some time now, I think.

**Comment by HuanhuanSunMSFT** [comment]
@shin- Probably I'm referring to this [issue](https://github.com/docker/compose/issues/4416). If I have some Chinese chars in the path, we hit the issue.

**Comment by shin-** [comment]
@HuanhuanSunMSFT Yes - #4122 and #4416 are still open. They're separate issues.

**Pull request merged**